### ISSUE TITLE
Use map instead of list for data_disk definition

### ISSUE THREAD

**Issue by barryib** [Terraform-VMWare-Modules/terraform-vsphere-vm#44]
Today, we use list for `data_disk_size_gb`, this mean, that we can't remove a disk without destroying them. By example, `["10", "5", "15"]`, removing the `"5"` will cause the destruction and recreation of two other disks.

To avoid that and enforce variable consistency, we can define data_disks as a map.

```hcl
data_disks = {
  disk1 = {
    size = 10,
    thin_provisioned = false
  },
  disk2 = {
    size = 5,
    thin_provisioned = true
  },
  disk3 = {
    size = 5
  }
}
```

We can also add another `data_disk_*` variables to this map to describe disks.

@Arman-Keyoumarsi What do you think ? We can propose a PR based on this proposal if it's ok for you.

**Comment by Arman-Keyoumarsi** [comment]
Good point. 

The change will definitely breaks backwards compatibility and up the release number. We have another PR waiting for merge which breaks backwards compatibility too. 

Let's do a PR, we can combine several major changes and do a full review.

**Comment by barryib** [comment]
Great. We'll push something in the next couple of day.

**Comment by achachw** [comment]
I finally pushed a new PR as proposal to use map for additional data disks and for network interfaces.
what do you thinks about it @Arman-Keyoumarsi  ?

**Comment by Arman-Keyoumarsi** [comment]
Already Implemented in v2.0.0
